Broken LabelledSeparator

#293 broke `LabelledSeparator`. No tests in orange-widget-base covered this, but orange3 tests fail with:

```
  File "/Users/janez/orange3/Orange/widgets/data/tests/test_owcontinuize.py", line 823, in test_contmodel
    self.assertIsNone(ind.data())
AssertionError: (None, '', False) is not None
```

`LabelledSeparator` is derived from `SeparatorItem` and adds a label. We want it to be a `Separator` to inherit other behaviour, but it contains additional data that is lost because `PyListModel.data` no longer returns after #293.

The easiest fix would be to replace `if isinstance(item, SeparatorItem): return None` with `if type(item) is SeparatorItem: return None`.

@ales-erjavec, would this be OK or should we do something more elaborate?

--- a/.github/workflows/run-tests-workflow.yml
+++ b/.github/workflows/run-tests-workflow.yml
@@ -44,15 +44,15 @@ jobs:
 
           - os: ubuntu-latest
             python-version: "3.12"
-            test-env: "PyQt6~=6.5.0 PyQt6-Qt6~=6.5.0 PyQt6-WebEngine~=6.5.0 PyQt6-WebEngine-Qt6~=6.5.0"
-
-          - os: ubuntu-latest
-            python-version: "3.13"
             test-env: "PyQt6~=6.8.0 PyQt6-Qt6~=6.8.0 PyQt6-WebEngine~=6.8.0 PyQt6-WebEngine-Qt6~=6.8.0"
 
           - os: ubuntu-latest
+            python-version: "3.13"
+            test-env: "PyQt6~=6.10.0 PyQt6-Qt6~=6.10.0 PyQt6-WebEngine~=6.10.0 PyQt6-WebEngine-Qt6~=6.10.0"
+
+          - os: ubuntu-latest
             python-version: "3.14"
-            test-env: "PyQt6~=6.9.0 PyQt6-Qt6~=6.9.0 PyQt6-WebEngine~=6.9.0 PyQt6-WebEngine-Qt6~=6.9.0"
+            test-env: "PyQt6~=6.11.0 PyQt6-Qt6~=6.11.0 PyQt6-WebEngine~=6.11.0 PyQt6-WebEngine-Qt6~=6.11.0"
 
           # macOS
           - os: macos-14
@@ -65,15 +65,15 @@ jobs:
 
           - os: macos-latest
             python-version: "3.12"
-            test-env: "PyQt6~=6.5.0 PyQt6-Qt6~=6.5.0 PyQt6-WebEngine~=6.5.0 PyQt6-WebEngine-Qt6~=6.5.0"
-
-          - os: macos-latest
-            python-version: "3.13"
             test-env: "PyQt6~=6.8.0 PyQt6-Qt6~=6.8.0 PyQt6-WebEngine~=6.8.0 PyQt6-WebEngine-Qt6~=6.8.0"
 
           - os: macos-latest
+            python-version: "3.13"
+            test-env: "PyQt6~=6.10.0 PyQt6-Qt6~=6.10.0 PyQt6-WebEngine~=6.10.0 PyQt6-WebEngine-Qt6~=6.10.0"
+
+          - os: macos-latest
             python-version: "3.14"
-            test-env: "PyQt6~=6.9.0 PyQt6-Qt6~=6.9.0 PyQt6-WebEngine~=6.9.0 PyQt6-WebEngine-Qt6~=6.9.0"
+            test-env: "PyQt6~=6.11.0 PyQt6-Qt6~=6.11.0 PyQt6-WebEngine~=6.11.0 PyQt6-WebEngine-Qt6~=6.11.0"
 
           # Windows
           - os: windows-2022
@@ -90,15 +90,15 @@ jobs:
 
           - os: windows-latest
             python-version: "3.12"
-            test-env: "PyQt6~=6.5.0 PyQt6-Qt6~=6.5.0 PyQt6-WebEngine~=6.5.0 PyQt6-WebEngine-Qt6~=6.5.0"
-
-          - os: windows-latest
-            python-version: "3.13"
             test-env: "PyQt6~=6.8.0 PyQt6-Qt6~=6.8.0 PyQt6-WebEngine~=6.8.0 PyQt6-WebEngine-Qt6~=6.8.0"
 
           - os: windows-latest
+            python-version: "3.13"
+            test-env: "PyQt6~=6.10.0 PyQt6-Qt6~=6.10.0 PyQt6-WebEngine~=6.10.0 PyQt6-WebEngine-Qt6~=6.10.0"
+
+          - os: windows-latest
             python-version: "3.14"
-            test-env: "PyQt6~=6.9.0 PyQt6-Qt6~=6.9.0 PyQt6-WebEngine~=6.9.0 PyQt6-WebEngine-Qt6~=6.9.0"
+            test-env: "PyQt6~=6.11.0 PyQt6-Qt6~=6.11.0 PyQt6-WebEngine~=6.11.0 PyQt6-WebEngine-Qt6~=6.11.0"
 
     steps:
       - uses: actions/checkout@v4

--- a/orangewidget/utils/itemmodels.py
+++ b/orangewidget/utils/itemmodels.py
@@ -503,7 +503,8 @@ class PyTableModel(AbstractSortTableModel):
 
 
 class SeparatorItem:
-    pass
+    def __str__(self):
+        return ""
 
 
 class LabelledSeparator(SeparatorItem):

--- a/orangewidget/utils/itemmodels.py
+++ b/orangewidget/utils/itemmodels.py
@@ -596,18 +596,14 @@ class PyListModel(QAbstractListModel):
         return 0 if parent.isValid() else 1
 
     def data(self, index, role=Qt.DisplayRole):
-        if not self._is_index_valid(index):
-            return None
         row = index.row()
-        if role in [self.list_item_role, Qt.EditRole]:
-            item = self[row]
-            if isinstance(item, SeparatorItem):
-                return None
-            return item
-        elif (role == Qt.AccessibleDescriptionRole and
-              isinstance(self[row], SeparatorItem)):
-            return 'separator'
-        else:
+        if role in [self.list_item_role, Qt.EditRole] \
+                and self._is_index_valid(index):
+            return self[row]
+        elif self._is_index_valid(row):
+            if isinstance(self[row], SeparatorItem) \
+                    and role == Qt.AccessibleDescriptionRole:
+                return 'separator'
             return self._other_data[row].get(role, None)
 
     def itemData(self, index):

--- a/orangewidget/utils/tests/test_itemmodels.py
+++ b/orangewidget/utils/tests/test_itemmodels.py
@@ -660,8 +660,12 @@ class TestPyListModel(unittest.TestCase):
         model += [1, model.Separator]
         model.extend([1, model.Separator])
         for i in range(len(model)):
-            self.assertIs(model.flags(model.index(i)) == Qt.NoItemFlags,
-                          i % 2 != 0, f"in row {i}")
+            if i % 2:
+                midx = model.index(i)
+                self.assertIs(midx.data(Qt.DisplayRole), model.Separator)
+                self.assertEqual(midx.data(Qt.AccessibleDescriptionRole),
+                                 "separator")
+                self.assertEqual(midx.flags(), Qt.NoItemFlags)
 
     def test_separator_display_text(self):
         delegate = QStyledItemDelegate()

--- a/orangewidget/utils/tests/test_itemmodels.py
+++ b/orangewidget/utils/tests/test_itemmodels.py
@@ -6,9 +6,10 @@ from unittest.mock import patch, Mock
 
 import numpy as np
 
-from AnyQt.QtCore import Qt, QModelIndex, QRect
+from AnyQt.QtCore import Qt, QModelIndex, QRect, QLocale
 from AnyQt.QtTest import QSignalSpy
 from AnyQt.QtGui import QPalette, QFont
+from AnyQt.QtWidgets import QStyledItemDelegate
 
 from orangewidget.utils.itemmodels import \
     AbstractSortTableModel, PyTableModel, PyListModel, \
@@ -661,6 +662,10 @@ class TestPyListModel(unittest.TestCase):
         for i in range(len(model)):
             self.assertIs(model.flags(model.index(i)) == Qt.NoItemFlags,
                           i % 2 != 0, f"in row {i}")
+
+    def test_separator_display_text(self):
+        delegate = QStyledItemDelegate()
+        self.assertEqual(delegate.displayText(PyListModel.Separator, QLocale()), "")
 
 
 class TestSeparatedListDelegate(unittest.TestCase):

--- a/orangewidget/utils/tests/test_itemmodels.py
+++ b/orangewidget/utils/tests/test_itemmodels.py
@@ -659,11 +659,8 @@ class TestPyListModel(unittest.TestCase):
         model += [1, model.Separator]
         model.extend([1, model.Separator])
         for i in range(len(model)):
-            if i % 2:
-                midx = model.index(i)
-                self.assertIs(midx.data(Qt.DisplayRole), None)
-                self.assertEqual(midx.data(Qt.AccessibleDescriptionRole), "separator")
-                self.assertEqual(midx.flags(), Qt.NoItemFlags)
+            self.assertIs(model.flags(model.index(i)) == Qt.NoItemFlags,
+                          i % 2 != 0, f"in row {i}")
 
 
 class TestSeparatedListDelegate(unittest.TestCase):

